lib/compress/zstd_opt.c

```diff
@@ -589,17 +606,47 @@ U32 ZSTD_insertBtAndGetAllMatches (
                      (ip+mlen == iLimit) ) {  /* best possible length */
                     ms->nextToUpdate = current+1;  /* skip insertion */
                     return 1;
-    }   }   }   }
+                }
+            }
+        } else if (dictMode == ZSTD_dictMatchState) {
+            /* should we perform this search even if the working ctx search
+             * found a match? */
+            U32 const hashLog3 = ms->hashLog3;
+            size_t const hash3 = ZSTD_hash3Ptr(ip, hashLog3);
+            U32 const dmsMatchIndex3 = dms->hashTable3[hash3];
```

Comment:
This hash table isn't actually populated in the dictionaries currently, which is causing the segfault you're seeing.

Reply:
If you add it, make sure to update https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L1300. Not sure if it is worth adding it though.